src/steps/malops/converters.ts

```diff
@@ -14,8 +35,85 @@ export function createMalopEntity(data: Malop) {
+          mapMalopSimpleFieldValue(
+            data.simpleValues.rootCauseElementCompanyProduct,
+          ) || undefined,
+        malopActivityTypes: mapMalopSimpleFieldValue(
+          data.simpleValues.malopActivityTypes,
+        ),
+        malopStartTime: mapMalopSimpleFieldValue(
+          data.simpleValues.malopStartTime,
+        )
+          ? parseInt(
+              mapMalopSimpleFieldValue(
+                data.simpleValues.malopStartTime,
+              ) as string,
+              10,
+            )
+          : undefined,
+        detectionType: mapMalopSimpleFieldValue(
+          data.simpleValues.detectionType,
+        ),
+        elementDisplayName: mapMalopSimpleFieldValue(
+          data.simpleValues.elementDisplayName,
+        ),
+        creationTime: mapMalopSimpleFieldValue(data.simpleValues.creationTime)
+          ? parseInt(
+              mapMalopSimpleFieldValue(
+                data.simpleValues.creationTime,
+              ) as string,
+              10,
+            )
+          : undefined,
+        isBlocked: mapMalopSimpleFieldValue(data.simpleValues.isBlocked),
+        rootCauseElementTypes: mapMalopSimpleFieldValue(
+          data.simpleValues.rootCauseElementTypes,
+        ),
+        rootCauseElementNames: mapMalopSimpleFieldValue(
+          data.simpleValues.rootCauseElementNames,
+        ),
+        malopLastUpdateTime: mapMalopSimpleFieldValue(
+          data.simpleValues.malopLastUpdateTime,
+        ),
+        allRansomwareProcessesSuspended: mapMalopSimpleFieldValue(
+          data.simpleValues.allRansomwareProcessesSuspended,
+        ),
+        managementStatus: mapMalopSimpleFieldValue(
+          data.simpleValues.managementStatus,
+        ),
+        closeTime: mapMalopSimpleFieldValue(data.simpleValues.closeTime)
+          ? parseInt(
+              mapMalopSimpleFieldValue(data.simpleValues.closeTime) as string,
+              10,
+            )
+          : undefined,
+        // In general, whenever we know some value can be null, we'd want to set it as undefined instead
+        closerName:
+          mapMalopSimpleFieldValue(data.simpleValues.closerName) || undefined,
+        // elementValues seem useful however they're not primitive types and neither arrays of primitive types
+        // which means we couldn't just map them like above, it'd be best if we leave them for later and check with J1
+        // sometimes this requires us to create new entities (that don't map to any of the endpoint API calls) but
+        // that we can connect with this entity
+        suspicions: data.suspicions || undefined,
+        suspicionCount: data.suspicionCount,
+        isMalicious: data.isMalicious,
+        malopPriority: data.malopPriority || undefined,
+        // Required fields for class "Assessment"
+        category: 'Risk Assessment',
+        summary:
+          data.isMalicious === false ? 'Is not malicious' : 'Is malicious',
+        internal: true,
```

Comment:
Is this something we do for all entities usually? Or is this something specific for this case? A bit confused since I went through the WPEngine integration that Paul worked on and didn't come across such a thing.

Reply:
Are you referring to the fact that I had to create that `mapMalopSimpleFieldValue` function or the fact that I introduced all the fields?

If former - if the mapping can't be made directly (mapping can be made directly if the target field is a primitive type or an array of primitive types), we need to resort to something like this.

If latter - yes we usually want most of the fields to land on our entities. Granted if we're sure that some field doesn't make sense or isn't relevant or isn't necessary - we don't want to port it over, though it's usually hard to decide if something truly is useless (there are a lot of customers and a lot of use-cases) so going with "broad strokes" initially is the safest approach.